third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -3622,7 +3622,7 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
         }
         let mut function_data: BTreeMap<FunId, FunctionData> = Default::default();
         for (name, entry) in &self.parent.fun_table {
-            if name.module_name != self.module_name {
```

Comment:
For scripts in two different files, since the module name are both "<SELF>", the check will fail. As a result, one script function will be added to multiple modules, leading to the error in #10992.